### DISCUSSION TITLE
Adonis v5 not outputing HTML Tags

### DISCUSSION THREAD

**Discussion by ekimkael** [adonisjs/core#4353] [Help] [answered]
Hello guys,
Hope you are doing well.

I'm using Adonisjs version 5. I escaped text from a WYSIWYG so I have this saved in my database:
`"lt;p&gt;We used 0 value for height and overflow kept to hide. We did so to bring smoothness while toggle (hide &amp;amp; show) text contents. Keep in mind that&amp;nbsp;&lt;a href=&quot;https:&#x2F;&#x2F;www.w3schools.com&#x2F;cssref&#x2F;pr_class_display.asp&quot; target=&quot;_blank&quot; rel=&quot;noopener noreferrer&quot;&gt;display none property&lt;&#x2F;a&gt;&amp;nbsp;will also work here, but CSS transition attribute (for smoothness) is not compatible with it.&lt;&#x2F;p&gt;\n&lt;p&gt;Now, we&amp;rsquo;ll design the visible area (heading of text contents). You can use custom colors and other CSS styles for this according to your needs.&lt;&#x2F;p&gt;"`

When I render I use the [safe method](https://v5-docs.adonisjs.com/reference/views/globals/safe) but it is rendering this:
<img width="1440" alt="Screenshot 2024-01-25 at 11 44 49" src="https://github.com/adonisjs/core/assets/10372405/7b3502d3-40d0-4672-8f65-7141bbc826f1">

Can someone help me?

**Comment by thetutlage** [comment, marked as answer]
The `safe` method does not unescape the escaped HTML. It is used to output HTML as it is (meaning do not escape it).

What you are looking for is an unescape method. You can use https://www.npmjs.com/package/he package and use it to `escape` and `unescape` both.

**Reply by ekimkael** [reply]
Thank you, it works
